modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectDefinitionLocalServiceTest.java

```diff
@@ -1126,6 +1128,196 @@ public void testEnableAccountEntryRestricted() throws Exception {
 		}
 	}
 
+	@Test
+	public void testEnableAccountEntryRestrictedForNondefaultStorageType()
+		throws Exception {
+
+		// Enabling account restriction to a custom salesforce object definition
+
+		ObjectDefinition objectDefinition1 =
+			_objectDefinitionLocalService.addCustomObjectDefinition(
+				TestPropsValues.getUserId(), false, false,
+				LocalizedMapUtil.getLocalizedMap("Able"), "Able", null, null,
+				LocalizedMapUtil.getLocalizedMap("Ables"),
+				ObjectDefinitionConstants.SCOPE_COMPANY,
+				ObjectDefinitionConstants.STORAGE_TYPE_SALESFORCE,
+				Collections.emptyList());
+
+		ObjectField objectField1 = ObjectFieldUtil.addCustomObjectField(
+			new TextObjectFieldBuilder(
+			).userId(
+				TestPropsValues.getUserId()
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).name(
+				StringUtil.randomId()
+			).objectDefinitionId(
+				objectDefinition1.getObjectDefinitionId()
+			).required(
+				true
+			).build());
+
+		objectDefinition1 =
+			_objectDefinitionLocalService.
+				enableAccountEntryRestrictedForNondefaultStorageType(
+					objectField1);
+
+		Assert.assertTrue(objectDefinition1.isAccountEntryRestricted());
+		Assert.assertTrue(
+			objectDefinition1.getAccountEntryRestrictedObjectFieldId() > 0);
+		Assert.assertFalse(objectDefinition1.isSystem());
+		Assert.assertEquals(
+			objectDefinition1.getStorageType(),
+			ObjectDefinitionConstants.STORAGE_TYPE_SALESFORCE);
+
+		_objectDefinitionLocalService.deleteObjectDefinition(objectDefinition1);
+
+		// Switching account restricted field
+
+		ObjectDefinition objectDefinition2 =
+			_objectDefinitionLocalService.addCustomObjectDefinition(
+				TestPropsValues.getUserId(), false, false,
+				LocalizedMapUtil.getLocalizedMap("Able"), "Able", null, null,
+				LocalizedMapUtil.getLocalizedMap("Ables"),
+				ObjectDefinitionConstants.SCOPE_COMPANY,
+				ObjectDefinitionConstants.STORAGE_TYPE_SALESFORCE,
+				Collections.emptyList());
+
```

Comment:
It is not usual to add new variables in the test, like: objectDefinition1, objectDefinition2, ... . We can only add more than one entity if they need to be created together in order to perform the Assert.

In this case, I think the ObjectDefinition variables are independent. Then, we should only override the existing variable instead of deleting and creating others. So, we would only need to delete it at the end of the test.

This same comment can be applied to the ObjectField variables.